### ISSUE TITLE
VBoxHeadless CPU

### ISSUE THREAD

**Issue by baroneddy** [kalabox/kalabox#1340]
Kbox version: v0.12.0-beta1 on El Capitan on a Mid-2012 Macbook pro
Previous usage: second time installing Kbox but first time really starting to use it.

He there (again, second time today :) ),

I have a few questions regarding virtualbox. Disclaimer: I am starting with all the servers side stuff so I might be saying crazy things.

So, when my Kalabox is started, my CPU is getting crazy, mostly because of VBoxHeadless. I have 2 questions around that:
- I understood that Docker released their last version where Docker don't need a virtual box any more. Is it planned to use this Kalabox?
- One the the popular solution that I found to reduce CPU of vbox for people using vagrant is to set $enable_serial_logging to false (https://github.com/Varying-Vagrant-Vagrants/VVV/issues/694). Any chance we can do something around that with Kalabox?

The reason for that is simple, when Kalabox is on, my computer gets really slow (all new el capitan install after a format), if I don't find a solution for that, I might need to find another solution, which I really don't want!

Thanks for your help and for all the work!

Ed.

**Comment by pirog** [comment]
Hey @baroneddy,

You are likely going to want to have at least 8GB RAM and possibly a newer processor to get a performant experience with Kalabox as it is currently constituted. 

Regarding your specific questions.

1. Yes! Docker is working on a "native" docker solution for Mac and Windows. It's currently in a private beta but we hope to switch to this as soon as it's ready to go. 
2. Do you know how this settings manifests on the VB side of things? IE is this a VB setting that Vagrant manages or is this Vagrant specific? 

Hopefully we can get this working nice and good for you!

**Comment by pirog** [comment]
Closing this in favor of #1368

**Comment by zanami** [comment]
I'm trying kalabox from time to time to see if it's performance getting better. The recent kalabox beta1 with one D8 application (switched off BTW) makes my old MBP15 (8Gb i5) fans constantly spin. VBoxHeadless and Syncthing together responsible for at least 80% CPU usage.
Other than that, it's the easiest solution for a drupal dev/builder I could find. Let's hope 'native' Docker/xhyve will solve it.
Just a comment.

**Comment by pirog** [comment]
We have similar hopes.

**Comment by wildkatana** [comment]
Thought I'd jump in here and say that I'm experiencing the 100% CPU issue as well on my Mac OSX  El Capitan i7 16GB RAM. I can't figure out where to put the enable_serial_logging = FALSE which has been suggested as a fix.

**Comment by pirog** [comment]
@wildkatana are you using the latest version?

**Comment by wildkatana** [comment]
@pirog I was using Version 0.13.0-beta.2 (0.13.0-beta.2). I tried updating to beta4 and restarting, but it is still having the same issue :(

**Comment by pirog** [comment]
@wildkatana hmmm yeah. earlier versions that used syncthing instead of unison had this issue but i havent seen any reports of newer versions. 

can you provide any additional details? 

im also not super familiar with the fix that is mentioned here. that seems a vagrant specific thing.

**Comment by wildkatana** [comment]
I don't recall having this issue when I used earlier versions, but I was also on earlier versions of Mac and I wasn't really paying as much attention then either.

I see the VBoxHeadless process start eating 100% of the CPU right after starting the Kalabox app itself, even if I haven't started running any of the apps yet. When I run the apps, they work fine, and it is all snappy, but the process is consistently eating 100% of the CPU. When I quit Kalabox, the process goes away.

It does seem like a Vagrant thing to me as well. Can you give me any pointers about where I should update my config for setting $enable_serial_logging to false? Is there a specific place I should use within Kalabox's config files? Or should I do it elsewhere?

**Comment by wildkatana** [comment]
I'm still hitting up against this issue. I tried a few things that I read online like disabling I/O APIC and Nested Paging, but that didn't work.

I did a lot of messing around with the settings seeing if anything stuck. One thing that I did notice that helped was removing the /Users shared folder. But this broke Kalabox.

So it seems like it is definitely something with the file system. Perhaps symlinks... I'll investigate more and post my findings here.

**Comment by pirog** [comment]
@wildkatana do you remember which version was "ok"? If anything i know that newer versions generally IMPROVED performance especially after we switching syncthing out for unison. 

For clarification kalabox does not use vagrant. So i dont think doing optimization there (unless you are running a vagrant box simultaneously with kalabox) is going to do much. 

Do you only see the CPU usage when you have an app started or does it seem to be high while idle as well?

**Comment by wildkatana** [comment]
@pirog Honestly I wasn't paying too much attention back then and it is hard to say that the issue wasn't present then as well since I only used Kalabox lightly due to other issues which have since been fixed.

My guess right now is that Syncthing may just have issues with the number of files on my disk. I have a 1TB drive and it is full of millions of files from all the random distros and things I download. Is there any reason why we share the entire Users directory? Maybe making that more precise would help?

Thanks for clarifying about Vagrant, no wonder I couldn't find the Vagrantfile. 

I see the VBoxHeadless start up right when Kalabox starts, and after spinning up (during which it is 200-300% CPU) it then settles on 100% CPU and doesn't go down. This is even with all of the apps within Kalabox turned off. Turning them on doesn't seem to affect it much. It's just a constant drain while Kalabox is open. Quitting Kalabox makes the VBoxHeadless go away.

**Comment by pirog** [comment]
@wildkatana sharing in the entire users directory is an upstream restriction imposed on us by `docker-machine` so that is unlikely to change until we move over to the new Docker beta (waiting on better-than-nfs file sharing there first). VB file sharing is a network mount so while idle i don't think it should be taking up a bunch of CPU so i dont think this is the source of the CPU usage.

I think the CPU issue you are seeing is likely from unison (which can run even if your app is stopped) https://github.com/kalabox/kalabox/issues/1544

**Comment by wildkatana** [comment]
I think you're right @pirog, I left my Kalabox running for a few hours and eventually the VBoxHeadless tapered off. I suspect it was just busy syncing files for a long time. Once they were all synced, the CPU usage dropped off. It started up again once I started an app though, but didn't go as high (about 40%). I'll follow that other thread and see what comes of it. Seems like there are still efficiency and performance gains to be made when syncing files.

**Comment by pirog** [comment]
@wildkatana sadly im not sure what additional gains can be made beyond restricting the set of files that are synced as per the docs.

**Comment by MrPaulDriver** [comment]
Just chiming in to say that I too am finding that vboxheadless is using more than 100% cpu. I noticed this because my Macbook Pro (16GB) was very hot to touch. I am using Kalabox version 2.0.1

**Comment by pirog** [comment]
@MrPaulDriver its likely worth waiting for the new 2.1 version to be posted. this uses a completely new backend that resolves a lot of the problems you are describing.
